### PR TITLE
Fix Redundant bound check for span.Length == 0 pattern

### LINKED ISSUE #67270: Redundant bound check for span.Length == 0 pattern
Found in https://github.com/dotnet/runtime/pull/66865#issuecomment-1081224071 

Simplified repro (from `String.JoinCore`):
```csharp
string Test1(ReadOnlySpan<string> values)
{
    if (values.Length <= 1)
        return values.Length == 0 ? string.Empty : values[0];
    return null;
}

string Test2(ReadOnlySpan<string> values)
{
    if (values.Length <= 1)
        return 0 >= (uint)values.Length ? string.Empty : values[0];
    return null;
}
```
codegen: [sharplab.io](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgYQYG8aHunyAGBgBUYuDOQAUAJRjYAJgHkAdgBsAngGUADtkUAeYvwB8DAG7ZlAVxEBKLj07UeThgEsAZg3FnLIgHQAZGEUAcwwACwZdAF4GcltHZ0TiAHZTcytcAKDQiKiYgQB+Xj5fAFF8TQxVBhA0n1wAbT4AXQBuOyTUxQtlZXaE7gBfGg7ioREMUikZBRUNbT0DPmNvDPinB0Tud09Vv0CQ8MiYuNGtplSBQxjxCxdFDGs9zIOchiKlsoqqmrqMpraZx4KQY3V6/Scw2ogyAA=)

`Test1` has a redundant rangecheck.

category:cq
theme:bounds-checks
skill-level:intermediate
cost:small
impact:small

### PR DESCRIPTION
**Pull request by EgorBo**
Closes https://github.com/dotnet/runtime/issues/67270

### CODE REVIEW

**Comment by EgorBo** [comment]
PTAL @jakobbotsch @AndyAyersMS cc @dotnet/jit-contrib
[Diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=650529&view=ms.vss-build-web.run-extensions-tab) aren't too big.

tldr: when we merge assertions (in this case, for Array.Length) we currently don't do a good job at properly tighten them from multiple assertions, this PR fixes this.

**Review by jakobbotsch — APPROVED** [review]
